src/compile/model.ts

```diff
@@ -169,12 +183,12 @@ export abstract class Model {
     // FIXME: write assembleScales() in scale.ts that
     // help assemble scale domains with scale signature as well
     return flatten(vals(this.component.scale).map((scales: ScaleComponents) => {
-      let arr = [scales.main];
+      let arr = [renameScaleData(this, scales.main)];
       if (scales.colorLegend) {
-        arr.push(scales.colorLegend);
+        arr.push(renameScaleData(this, scales.colorLegend));
```

Comment:
looks like `renameScaleData` is only used here.  Should it be just a private method for model?